File: bms/go/ach/moov_io_ach_converters_formatSimpleTime/moov_io_ach_converters_formatSimpleTime.go
package test

import (
	"strings"
	"unicode/utf8"
)

var (
	moov_io_ach_stringZeros map[int]string = moov_io_ach_populateMap(94, "0")
)

const ()

func (c *moov_io_ach_converters) formatSimpleTime(s string) string {
	if s == "" {
		return c.stringField(s, 4)
	}
	return s
}

func (c *moov_io_ach_converters) stringField(s string, max uint) string {
	ln := uint(utf8.RuneCountInString(s))
	if ln > max {
		return s[:max]
	}

	// Pad with preallocated string
	m := int(max - ln)
	pad, exists := moov_io_ach_stringZeros[m]
	if exists {
		return pad + s
	}
	// slow path
	return strings.Repeat("0", m) + s
}

func moov_io_ach_populateMap(max int, zero string) map[int]string {
	out := make(map[int]string, max)
	for i := 0; i < max; i++ {
		out[i] = strings.Repeat(zero, i)
	}
	return out
}

type moov_io_ach_converters struct{}

File: translations_only/fail/go/ach/moov_io_ach_converters_formatSimpleTime/mistral/Restart/moov_io_ach_converters_formatSimpleTime.rs

use std::collections::HashMap;
use std::fmt;

struct MoovIoAchConverters;

const MOOV_IO_ACH_ZERO: &str = "0";

fn moov_io_ach_populate_map(max: usize, zero: &str) -> HashMap<usize, String> {
let mut out = HashMap::new();
for i in 0..max {
out.insert(i, zero.repeat(i));
}
out
}
